docs/guides/building_on_macos.md

```diff
@@ -0,0 +1,65 @@
+# Building DSDA-Doom on MacOS
+This is a basic guide for building DSDA-Doom for a x86_64 or arm64 MacOS target using brew. 
+## Configure brew
+[brew](https://brew.sh) is a package manager for MacOS and Linux. we will use it to download everything we need to build DSDA-Doom.
+
+To install it we need to run:
+```
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+```
+On x86_64 machines, brew will be installed in `/usr/local/homebrew`.
+
+On arm64 machines, brew will be installed in `/opt/homebrew`.
+## Install Build Dependencies
+Install cmake, SDL2 and additional dependencies for DSDA-Doom:
+```
+brew install cmake pcre sdl2 sdl2_net sdl2_image sdl2_mixer fluid-synth portmidi libmad dumb libvorbis
```

Comment:
Do we still need sdl2_net?

Reply:
On the prboom+ guide for windows they still install it, so I thought it was necessary. But no, we dont need it